rewrite-maven/src/main/java/org/openrewrite/maven/UpdateMavenProjectPropertyJavaVersion.java

```diff
@@ -92,14 +92,20 @@ public Xml.Document visitDocument(Xml.Document document, ExecutionContext ctx) {
                     for (Plugin plugin : getResolutionResult().getParent().getPom().getPlugins()) {
                         if (plugin.getGroupId().equals("org.apache.maven.plugins") && plugin.getArtifactId().equals("maven-compiler-plugin") && plugin.getConfiguration() != null) {
                             for (String property : JAVA_VERSION_PROPERTIES) {
-                                if (getResolutionResult().getPom().getProperties().get(property) != null) {
+                                String propValue = getResolutionResult().getPom().getProperties().get(property);
```

Comment:
I think this `getValue()` is what you were looking for:
```suggestion
                                String propValue = getResolutionResult().getPom().getValue(getResolutionResult().getPom().getProperties().get(property));
```

Reply:
Although that won't help with updating that "deepest" property. But I am also unsure if we really want that, as that property could also have other purposes. So I am not really sure what we should do here. How do other recipes handle this problem?

What are your thoughts here, @timtebeek ? Maybe it is after all best to leave such property references alone and then require the user to have a recipe to update those separately.

Reply:
Not sure I have much more insights to share here; if indeed a property only refers to another one I think it could help if we upgrade the nested property, or risk not compiling because the source level is not set correctly in say a compiler plugin.

What would happen here without any changes? Do we override `java.version` discarding the old reference? Or do we not make any change at all?

Reply:
Without this PR no changes are made. To me it is unclear which approach is best. Following a consistent pattern across recipes would be best, but maybe we don't have any other cases like this one.

Reply:
Without the change it will check these conventional properties https://github.com/openrewrite/rewrite/blob/20dc764c508a7be8d1c5e197f39fe7508a510c2e/rewrite-maven/src/main/java/org/openrewrite/maven/UpdateMavenProjectPropertyJavaVersion.java#L37-L45 but if any of them refer to another property (that is not coincidentally part of that list) nothing will change.